Controller/controller.proto

```diff
@@ -0,0 +1,31 @@
+syntax = "proto2";
+
+message ControllerConfig {
+    optional ProbeConfigs probes = 1;
```

Comment:
I know may seem counter-intuitive, but the best practice is to not use "required".

It makes the proto brittle and hard to change later.  Let's say you have a client and server, the proto has a field marked required.  Then later on you decide to re-write your proto, and the field isn't needed anymore.  And you can't really safely change it without downtime.  If the client tries to serialize a proto without the field, it won't work.  If the server tries to deserialize a proto without the field, it won't work.

go/required-considered-harmful  - hopefully you have access to that.

The workaround is to mark things as optional, and then enforce the required-ness in code.  I.e. The server should reject requests w/o necessary fields.

Reply:
addressed in 6a760db